src/Builder/CliMenuBuilder.php

```diff
@@ -194,6 +202,14 @@ public function addSubMenu(string $text, \Closure $callback) : self
             $menu->setStyle($this->menu->getStyle());
         }
 
+        if (!$menu->getCheckboxStyle()->hasChangedFromDefaults()) {
+            $menu->setCheckboxStyle(clone $this->menu->getCheckboxStyle());
```

Comment:
Looking back the way we did this before: If do not modify any child menu's style. We set the same instance of `MenuStyle` on the children. This way it's a universal theme and if you change something anywhere, it's changed everywhere. I'm particularly bothered but maybe we should be consistent and not clone. What do you think?

Reply:
All styles pointing to the same style object instance would prevent submenu styling.

The current `checkbox-item.php` example:

```
<?php

use PhpSchool\CliMenu\CliMenu;
use PhpSchool\CliMenu\Builder\CliMenuBuilder;
use PhpSchool\CliMenu\Style\CheckboxStyle;

require_once(__DIR__ . '/../vendor/autoload.php');

$itemCallable = function (CliMenu $menu) {
    echo $menu->getSelectedItem()->getText();
};

$menu = (new CliMenuBuilder)
    ->setTitle('Select a Language')
    ->addSubMenu('Compiled', function (CliMenuBuilder $b) use ($itemCallable) {
        $b->setTitle('Compiled Languages')
            ->addCheckboxItem('Rust', $itemCallable)
            ->addCheckboxItem('C++', $itemCallable)
            ->addCheckboxItem('Go', $itemCallable)
            ->addCheckboxItem('Java', $itemCallable)
            ->addCheckboxItem('C', $itemCallable)
        ;
    })
    ->addSubMenu('Interpreted', function (CliMenuBuilder $b) use ($itemCallable) {
        $b->setTitle('Interpreted Languages')
            ->modifyCheckboxStyle(function (CheckboxStyle $style) {
                $style->setMarkerOff('[○] ')
                    ->setMarkerOn('[●] ');
            })
            ->addCheckboxItem('PHP', $itemCallable)
            ->addCheckboxItem('Javascript', $itemCallable)
            ->addCheckboxItem('Ruby', $itemCallable)
            ->addCheckboxItem('Python', $itemCallable)
        ;
    })
    ->build();

$menu->open();
```

In this case I want the main menu, and the first submenu, to use the default theme, but I want the second submenu's theme modified and any children it has to also use that modified theme.

Reply:
All styles pointing to the same style object instance would prevent submenu styling.

The current `checkbox-item.php` example:

```
<?php

use PhpSchool\CliMenu\CliMenu;
use PhpSchool\CliMenu\Builder\CliMenuBuilder;
use PhpSchool\CliMenu\Style\CheckboxStyle;

require_once(__DIR__ . '/../vendor/autoload.php');

$itemCallable = function (CliMenu $menu) {
    echo $menu->getSelectedItem()->getText();
};

$menu = (new CliMenuBuilder)
    ->setTitle('Select a Language')
    ->addSubMenu('Compiled', function (CliMenuBuilder $b) use ($itemCallable) {
        $b->setTitle('Compiled Languages')
            ->addCheckboxItem('Rust', $itemCallable)
            ->addCheckboxItem('C++', $itemCallable)
            ->addCheckboxItem('Go', $itemCallable)
            ->addCheckboxItem('Java', $itemCallable)
            ->addCheckboxItem('C', $itemCallable)
        ;
    })
    ->addSubMenu('Interpreted', function (CliMenuBuilder $b) use ($itemCallable) {
        $b->setTitle('Interpreted Languages')
            ->modifyCheckboxStyle(function (CheckboxStyle $style) {
                $style->setMarkerOff('[○] ')
                    ->setMarkerOn('[●] ');
            })
            ->addCheckboxItem('PHP', $itemCallable)
            ->addCheckboxItem('Javascript', $itemCallable)
            ->addCheckboxItem('Ruby', $itemCallable)
            ->addCheckboxItem('Python', $itemCallable)
        ;
    })
    ->build();

$menu->open();
```

In this case I want the main menu, and the first submenu, to use the default theme, but I want the second submenu's theme modified and any children it has to also use that modified theme.

Reply:
Ok that's fine.